core/embed/rust/src/ui/util.rs

```diff
@@ -131,6 +131,18 @@ pub fn char_to_string(ch: char) -> ShortString {
     s
 }
 
+/// Returns the first two lines of text to be fit on a given length.
+/// When the text is too long to fit in two lines,
+/// the rest is just kept as part of the second line.
+pub fn split_two_lines(text: &str, text_font: Font, available_width: i16) -> (&str, &str) {
+    let p = text_font.longest_prefix_words(available_width, text);
+    if p == 0 { // we did not find a space to split on...
+        (&text[0..0], &text[0..])
+    } else {
+        (&text[..p + 1], &text[p + 2..])
```

Comment:
...and this is magically skipping a single ASCII space? that will fail if the longest prefix is the whole of the text.

(you'll probably want `str::trim` here)

Reply:
Indeed, https://github.com/trezor/trezor-firmware/pull/4226/commits/92302c1513c3aaf5c4f3634de07b9d3e65da183b#diff-49bfb5a6d75553b6b483d6d27558216874b122622f76dc4ebdcf835510e1bf40R145